src/coreclr/tools/aot/ILCompiler/Program.cs

```diff
@@ -420,13 +420,14 @@ private int Run(string[] args)
                 optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("popcnt");
                 optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("lzcnt");
 
-                // If AVX was enabled, we can opportunistically enable FMA/BMI
+                // If AVX was enabled, we can opportunistically enable FMA/BMI/VNNI
                 Debug.Assert(InstructionSet.X64_AVX == InstructionSet.X86_AVX);
                 if (supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX))
                 {
                     optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("fma");
                     optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("bmi");
                     optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("bmi2");
+                    optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("avxvnni");
```

Comment:
This is going to add `AVX2` to the list. Is that problematic?

My guess is no since we only use the `supportedInstructionSet` flags to determine things like `Vector<T>` length, but seeing as `AVX2` is already "missing" here, I thought I'd ask.

Reply:
Yeah, AVX2 is probably an omission. I don't think there's a reason why we wouldn't be able to support it under the runtime check.